client/src/js/services/receipts/ReceiptModal.js

```diff
@@ -0,0 +1,55 @@
+angular.module('bhima.services')
+.service('ReceiptModal', ReceiptModal);
+
+ReceiptModal.$inject = ['$uibModal', 'ReceiptService'];
+  
+/** 
+ * Receipts Modal Service 
+ *
+ * This service is responsible for combining receipt service data with the 
+ * receipts modal controller and providing a clean interface to be used within 
+ * module controllers. 
+ *  
+ * @todo Discuss how the render target PDF should be templated. Suggestion: 
+ *       - IF pdf   - use /src/services/receipts/modal/pdf.tmpl.html
+ *       - IF json  - specifiy individual target /partials/patient_invoice/receipt/...
+ *
+ * @module services/receipts/ReceiptModal
+ */
+function ReceiptModal(Modal, Receipts) { 
+  var service = this;
+  
+  var modalConfiguration = { 
+    templateUrl : '/js/services/receipts/modal/receiptModal.tmpl.html',
+    controller  : 'ReceiptModalController as ReceiptCtrl',
+    size        : 'md',
+    backdrop    : 'static',
+    animation   : false
+  };
+  
+  // expose available receipts
+  service.invoice = invoice;
+
+  function invoice(uuid) { 
+
+    /** @todo Discuss if these should be overridable from the controller or if the config should be set here */
+    var options = { 
```

Comment:
@jniles in response to your question on gitcolony.

>I am having trouble understanding how these options might be passed in.  Is this ReceiptModal generic to all receipts (cash, purchase, etc), or will it only do Patient Invoices?  How would these options get set if this were to be shared for multiple invoices? 



> I looked at https://github.com/IMA-WorldHealth/bhima-2.X/issues/258, but it doesn't seem to mention that plan.  Where else should I look?

The `ReceiptModal` service is currently the central point that configures both the data request and the view binding for any receipt - that means that the options set here influence both the query and the information passed to the receipt modal controller. 

With this layout this is the *only* file the developer should have to change to configure receipts on the client - the receipts modal controller is responsible for taking these configuration options and binding the view and the `ReceiptsService` acts as a testable layer between the server and the client.

To provide more context, with cash and purchase it would look as follows: 
**Receipt Modal Service**
```js 
function ReceiptModal(Modal, Receipts) { 
  var modalConfiguration = //...

  function invoice(uuid) {
    var options = //...
    
    // extends modal configuration with invoice options to allow view binding in the receipts modal
  }

  function cash(uuid) { 
    var options = //...
  }

  function purchase(uuid) { 
    var options = //...
  }
}
```

It's worth noting that all of the methods here share the same modal configuration so any changes to this would be applied to all receipts unless they overrode the default configuration. 

This is just the simplest (initial) use case. If anyone has any suggestions as to a better layout that would be great! @IMA-WorldHealth/local-contributors